apps/rush-lib/src/cli/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -124,9 +135,13 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
 
   /**
    * abstract
-   * Gets the resolved version number of a dependency for a specific temp project
+   * Gets the resolved version number of a dependency for a specific temp project.
+   * For PNPM, we can reuse the version that another project is using.
+   * Note that this function modifies the shrinkwrap data.
    */
-  protected getDependencyVersion(dependencyName: string, tempProjectName: string): string | undefined {
+  protected tryEnsureDependencyVersion(dependencyName: string,
+    tempProjectName: string,
```

Comment:
>string [](start = 21, length = 6)

The tempProjectName parameter is declared as `string | undefined` in the base class.  As I mentioned before, I'm pretty uncomfortable with overridden functions altering the type signature of the base implementation.  In my head it's too difficult to verify the correctness of that sort of design.